examples/intro_data_profiler.ipynb

```diff
@@ -0,0 +1,431 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2b7409c6",
+   "metadata": {},
+   "source": [
+    "In addition to reading the input data from multiple file types, DataProfiler allows the input data as a dataframe."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cb528718",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# run data profiler and get the report\n",
+    "my_dataframe = pd.DataFrame([[1, 2.0],[1, 2.2],[-1, 3]], columns=['col_int', 'col_float'])\n",
+    "profile = dp.Profiler(my_dataframe)\n",
+    "report  = profile.report(report_options={\"output_format\":\"compact\"})\n",
+    "\n",
+    "# Print the report\n",
+    "print(json.dumps(report, indent=4))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "273e8b63",
+   "metadata": {},
+   "source": [
+    "## Profiler options"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "88eaa485",
+   "metadata": {},
+   "source": [
+    "DataProfiler can run several selected components if needed. For example, if the users only want the statistics information, they may turn off the DataLabeler functionality. Below, let's remove the histogram and data labeler component while running DataProfiler."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9417a40c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "profile_options = dp.ProfilerOptions()\n",
+    "profile_options.set({\"histogram.is_enabled\": False,\n",
+    "                     \"data_labeler.is_enabled\": False,})\n",
+    "\n",
+    "profile = dp.Profiler(data, profiler_options=profile_options)\n",
+    "report  = profile.report(report_options={\"output_format\":\"compact\"})\n",
+    "\n",
+    "# Print the report\n",
+    "print(json.dumps(report, indent=4))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4a871d78",
+   "metadata": {},
+   "source": [
+    "## Data reader class"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b1963a8a",
+   "metadata": {},
+   "source": [
+    "DataProfiler can detect multiple file types including CSV, JSON, Parquet, AVRO, and text. The example below shows that it successfully detects data types from multiple categories regardless of the file extensions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7738c719",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# use data reader to read input data with different file types\n",
+    "csv_files = [\n",
+    "    \"../dataprofiler/tests/data/csv/aws_honeypot_marx_geo.csv\",\n",
```

Comment:
these are all super long, is it possible we create a variable for the common path and then add them on later? it would make this more readable and fit all the text more easily.

Reply:
added a data path